### PR TITLE
Refactor/gradle restructurization

### PR DESCRIPTION
**Pull request by ErikhPetrushynets**
In this PR implemented refactoring of station_manager project.

### CODE REVIEW

File: backend/station_manager/shared/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+dependencies {
+    // Use JUnit Jupiter for testing.
+    testImplementation("org.junit.jupiter:junit-jupiter:5.9.3")
+
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+
+    // This dependency is used by the application.
+    implementation("com.google.guava:guava:32.1.1-jre")
+    implementation("io.javalin:javalin:5.6.3")
+    implementation("org.hibernate.orm:hibernate-core:6.4.0.Final")
```
Comment (serozhenka) [inline backend/station_manager/shared/build.gradle.kts]: no need for this dependencies in shared

**Review by serozhenka — CHANGES_REQUESTED** [review]
Really good job, thanks. 3 things:
- move domain tests to shared
- simplify `generator_simulator` name
- change `.github/workflows/tests.yml` to run tests on subprojects (the latest step)

File: backend/station_manager/shared/src/main/java/com/StationManager/shared/domain/train_station/TicketOffice.java [hunk 2]
```diff
@@ -22,7 +22,7 @@ public class TicketOffice {
     private Direction direction;
     private Iterable<ServeRecord> transactions;
 
-    public final Queue<Event> events = new LinkedList<>();
+    public final Queue<com.StationManager.shared.domain.events.Event> events = new LinkedList<>();
```
Comment (serozhenka) [inline backend/station_manager/shared/src/main/java/com/StationManager/shared/domain/train_station/TicketOffice.java]: maybe import it first? and then just reference the `Event`?

Comment (ErikhPetrushynets) [reply backend/station_manager/shared/src/main/java/com/StationManager/shared/domain/train_station/TicketOffice.java]: I do that in last commit

Comment (ErikhPetrushynets) [reply backend/station_manager/shared/src/main/java/com/StationManager/shared/domain/train_station/TicketOffice.java]: commit*

**Review by serozhenka — APPROVED** [review]

File: backend/station_manager/shared/src/main/java/com/StationManager/shared/domain/train_station/TicketOffice.java (hunk 2 above)
Comment (serozhenka) [reply backend/station_manager/shared/src/main/java/com/StationManager/shared/domain/train_station/TicketOffice.java]: sorry, missed that

**Pull request merged**